hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/record/RecordListBuilder.java

```diff
@@ -253,19 +253,21 @@ private SingleTransactionRecordBuilderImpl doAddChild(
         }
 
         // Make sure we have not created so many that we have run out of slots.
-        final var childCount = childRecordBuilders.size();
-        final var consensusConfig = configuration.getConfigData(ConsensusConfig.class);
+        final int childCount = childRecordBuilders.size();
+        final ConsensusConfig consensusConfig = configuration.getConfigData(ConsensusConfig.class);
         if (childCount >= consensusConfig.handleMaxFollowingRecords()) {
             throw new HandleException(ResponseCodeEnum.MAX_CHILD_RECORDS_EXCEEDED);
         }
 
-        // The consensus timestamp of the first item in the child list is T+1, where T is the time of the user tx
-        final var parentConsensusTimestamp = userTxnRecordBuilder.consensusNow();
-        final var prevConsensusNow = childRecordBuilders.isEmpty()
-                ? userTxnRecordBuilder.consensusNow()
+        // The consensus timestamp of the first item in the child list is T+K (in nanoseconds),
+        // where T is the time of the user tx and K is the maximum number of "preceding" records
+        // defined for the current configuration.
+        final long maxPrecedingRecords = consensusConfig.handleMaxPrecedingRecords();
```

Comment:
Nit: we can use this variable already in the condition above

Reply:
If you mean line 258, that's a different value (following versus preceding).
Otherwise I'm a bit confused what condition is being referenced.

Reply:
Oh, you are right. Sorry, false alarm.